File: Laboratorio 1/Tp2/src/main/java/Fraccion.java

public class Fraccion {
    int numerador;
    int denominador;
    public Fraccion(int numerador, int denominador) {
        this.numerador = numerador;
        this.denominador = denominador;
    }
     @Override
    public String toString() {
        
        return numerador + "/" + denominador;
    }
     
public Fraccion(){
}
 

    public int getNumerador() {
        return numerador;
    }

    public int getDenominador() {
        return denominador;
    }

    public void setNumerador(int numerador) {
        this.numerador = numerador;
    }

    public void setDenominador(int denominador) {
        this.denominador = denominador;
    }
    
   
     
    public Fraccion sumarFracciones(Fraccion f1,Fraccion f2) {

        Fraccion f = new Fraccion(); 
        f.setDenominador(f1.getDenominador() * f2.getDenominador());
        f.setNumerador((f2.getDenominador() * f1.getNumerador()) + (f2.getNumerador() * f1.getDenominador()));
        
        return f;
    }
    public Fraccion restarFracciones(Fraccion f1, Fraccion f2){
      Fraccion f = new Fraccion();
       f.setDenominador(f1.getDenominador() * f2.getDenominador());
       f.setNumerador((f1.getNumerador() * f2.getDenominador()) - (f2.getNumerador() * f1.getDenominador()));
         
         
      return f;
}
    public Fraccion multiplicarFracciones(Fraccion f1, Fraccion f2){

       Fraccion f = new Fraccion();

       f.setNumerador(f1.getNumerador() * f2.getNumerador());
       f.setDenominador(f1.getDenominador() * f2.getDenominador());
       
       return f;       
}
    public Fraccion dividirFracciones(Fraccion f1, Fraccion f2){
        Fraccion f = new Fraccion();
        f.setNumerador(f1.getNumerador() * f2.getDenominador());
        f.setDenominador(f1.getDenominador() * f2.getNumerador());
        
return f;
}


    
}

File: Laboratorio 3/Unidad 1/Codigo/Tp_numero_2/Tp_numero_2/Fraccion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace Tp_numero_2
{
    internal class Fraccion{
       
        private int numerador;
        private int denominador;


        public Fraccion(int numerador, int denominador){
            if (denominador == 0)
            {
                throw new FraccionException("El denominador no puede ser cero");
            }

            this.numerador = numerador;
            this.denominador = denominador;
        }



        //public int GetNumerador() { 
        
        //return numerador;
        //}
        //public void SetNumerador(int numerador)
        //{
        //    this.numerador = numerador;
        //}

        //public int GetDenominador()
        //{

        //    return denominador;
        //}
        //public void SetDenominador(int denominador)
        //{
        //    this.denominador = denominador;
        //}

        public int Sumar() {
         
            return (numerador + denominador);
        }
        public int Restar() {

            return(numerador-denominador);
        }
        public int Multiplicar() { 
        return (numerador * denominador);
        }
        public int Dividir() { 

            return (numerador/denominador);
        }






    }
}
